dags/gcc_layers_pull.py

```diff
@@ -17,115 +17,83 @@
 except:
     raise ImportError("Cannot import DAG helper functions.")
 
-# Credentials - to be passed through PythonOperator
-# bigdata connection credentials
-bigdata_cred = PostgresHook("gcc_bot_bigdata")
-# On-prem server connection credentials
-ptc_cred = PostgresHook("gcc_bot")
-
-DAG_NAME = 'pull_gcc_layers'
 
-DAG_OWNERS  = Variable.get('dag_owners', deserialize_json=True).get(DAG_NAME, ["Unknown"])
+# the DAG runs at 7 am on the first day of January, April, July, and October
+def create_gcc_puller_dag(dag_id, default_args, name, conn_id):
+    @dag(
+        dag_id=dag_id,
+        default_args=default_args,
+        catchup=False,
+        tags=['gcc', name],
+        schedule='0 7 1 */3 *' #'@quarterly'
+    )
+    def gcc_layers_dag():
+        
+        @task()
+        def get_layers(name):
+            tables = Variable.get('gcc_layers', deserialize_json=True)
+            return tables[name]
+                    
+        @task() #add after 2.9.3 upgrade: map_index_template="{{ table_name }}"
+        def pull_layer(layer, conn_id):
+            #name mapped task (implement after 2.9.3 upgrade)
+            #context = get_current_context()
+            #context["table_name"] = layer[0]
+            #get db connection
+            conn = PostgresHook(conn_id).get_conn()
+            
+            #pull and insert layer
+            get_layer(
+                mapserver_n = layer[1].get("mapserver"),
+                layer_id = layer[1].get("layer_id"),
+                schema_name = layer[1].get("schema_name"),
+                is_audited = layer[1].get("is_audited"),
+                primary_key = layer[1].get("pk"),
+                con = conn
+            )
 
-DEFAULT_ARGS = {
-    'owner': ','.join(DAG_OWNERS),
-    'depends_on_past': False,
-    'start_date': pendulum.datetime(2022, 11, 3, tz="America/Toronto"),
-    'email_on_failure': False, 
-    'retries': 0,
-    'on_failure_callback': partial(task_fail_slack_alert, use_proxy=True)
-}
+            #refresh mat views as necessary
+            agg_sql = layer[1].get("agg")
+            if agg_sql is not None:
+                with conn.cursor() as cur:
+                    print(agg_sql)
```

Comment:
this for logging purpose?

Reply:
Oh I'll remove this, it was for testing (instead of execute).